### PR TITLE
[chore] build images for PRs

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-demo/issues/952

This should give more confidence when merging Renovate PRs (reference https://github.com/open-telemetry/opentelemetry-demo/pull/931).

### CODE REVIEW

File: .github/workflows/build-images.yml
Comment (Kielek) [inline .github/workflows/build-images.yml:120]: I think that push should be done only for release.yml not for checks.

Comment (pellared) [reply .github/workflows/build-images.yml:120]:
```
      push:
        description: Should the images be pushed
        default: false
```

on the top

+

```
push: ${{ inputs.push }}
``` a few lines below

**Comment by pellared** [comment]
https://github.com/open-telemetry/opentelemetry-demo/actions/runs/5398517652/jobs/9804417710?pr=955 is failing

However, I guess the same problem would happen during release and it is rather not related with this PR.

I do not think it should be a blocker for this PR.

**Comment by GaryPWhite** [comment]
> I do not think it should be a blocker for this PR.

I'm not an approver; but I will say that introducing a step that fails should absolutely block the PR to avoid other folks having to address the change. Who else would be suited to resolve the problem?

**Comment by GaryPWhite** [comment]
unless we can tell something got busted in the dependency pipeline it seems like the pipeline got borked somehow with this change, no?

**Comment by pellared** [comment]
> nless we can tell something got busted in the dependency pipeline it seems like the pipeline got borked somehow with this change, no?

Normally I would agree. However, these jobs are used for building the container images during "release" workflow. So right now it gives fast feedback that the release would not work 😉

**Comment by pellared** [comment]
> > I do not think it should be a blocker for this PR.
> 
> I'm not an approver; but I will say that introducing a step that fails should absolutely block the PR to avoid other folks having to address the change. Who else would be suited to resolve the problem?

Normally I would agree. However, these jobs are used for building the container images during "release" workflow. So right now it gives fast feedback that the release would not work 😉

**Comment by pellared** [comment]
> does this only build what's changed in a PR, or everything

Everything 😉

**Comment by pellared** [comment]
> we do need to fix whatever's broken in the build currently...

I created https://github.com/open-telemetry/opentelemetry-demo/issues/956 and also commented the line which needs to be uncommented https://github.com/open-telemetry/opentelemetry-demo/pull/955/commits/018b9ea120dc8d1cb6b3547afa0487a8695561d1

**Comment by pellared** [comment]
Transient error: https://github.com/open-telemetry/opentelemetry-demo/actions/runs/5540546109/jobs/10112777382

Can somebody rerun? 😉

**Comment by julianocosta89** [comment]
The build really takes some time 😢

**Review by austinlparker — APPROVED** [review]

**Comment by austinlparker** [comment]
I think we're just gonna have to deal with the build times, but I'm more concerned about bringing stuff in to main that doesn't build properly. PR lgtm, lets goooo

**Pull request merged**
